### PR TITLE
Fix dumpstruct for anonymous structures

### PR DESCRIPTION
**Pull request by Schamper**

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -23,32 +23,41 @@ def test_hexdump(capsys):
 
 
 def test_dumpstruct(capsys, compiled):
```
Comment (Miauwkeru) [inline tests/test_util.py]:
```suggestion
@pytest.mark.parametrize(
    "cdef",
    [
        """
    struct test {
        uint32 testval;
    };
    """,
        """
    struct test {
        struct {
            uint32 testval;
        };
    };
    """,
    ],
)
def test_dumpstruct(capsys, compiled: bool, cdef: str) -> None:
```
wouldn't this be a bit neater?

**Review by Miauwkeru — CHANGES_REQUESTED** [review]

File: tests/test_util.py (hunk 1 above)
Comment (Schamper) [reply tests/test_util.py]: I don't find this very readable, plus we may want to test more complex dumpstructs later that don't follow this simple uint32 test value.

Comment (Miauwkeru) [reply tests/test_util.py]: Fair, tho when running the test. if one of them fails you'd need to check which one if them fails first. If you separate them in different tests, you at least have an idea where to look at/which functionality broke.

as for the future dumpstruct tests, wouldn't you just add separate tests for those anyways? And if the check logic is the same, maybe move the check logic to its own function or something.

Comment (Schamper) [reply tests/test_util.py]: Put this in a separate test

**Review by Miauwkeru — APPROVED** [review]
LGTM

**Pull request merged**